pkg/storegateway/indexheader/index/symbols.go

```diff
@@ -100,6 +100,67 @@ func (s *Symbols) Lookup(o uint32) (sym string, err error) {
 	return sym, nil
 }
 
+func (s *Symbols) Reader() *SymbolsTableReader {
+	d := s.factory.NewDecbufAtUnchecked(s.tableOffset)
+	d.ResetAt(s.offsets[0])
+	return &SymbolsTableReader{
+		d: &d,
+		s: s,
+	}
+}
+
+type SymbolsTableReader struct {
+	// atSymbol is the sequence of the symbol in V2, and the offset of the symbol in V1
+	atSymbol uint32
+	d        *streamencoding.Decbuf
+	s        *Symbols
+}
+
+func (r *SymbolsTableReader) Close() error {
+	return r.d.Close()
+}
+
+func (r *SymbolsTableReader) Read(o uint32) (string, error) {
+	d := r.d
+	if err := d.Err(); err != nil {
+		return "", err
+	}
+
+	if o < r.atSymbol {
+		return "", fmt.Errorf("trying to reverse SymbolsTableReader, at %d requesting %d", r.atSymbol, o)
```

Comment:
Perhaps this would be clearer?

```suggestion
		return "", fmt.Errorf("trying to read symbol at earlier position, at %d requesting %d", r.atSymbol, o)
```